CMakeLists.txt

```diff
@@ -156,6 +161,7 @@ message(STATUS "Setting namespace to '${MATERIALX_NAMESPACE}'")
 set (MATERIALX_LIBNAME_SUFFIX "" CACHE STRING "Specify a suffix to all libraries that are built")
 
 mark_as_advanced(MATERIALX_BUILD_DOCS)
+mark_as_advanced(MATERIALX_BUILD_PYTHON_DOCS)
```

Comment:
This naming convention for this new option makes good sense, though it makes our original `MATERIALX_BUILD_DOCS` option seem ambiguous in its meaning.

Perhaps we should rename the original option to `MATERIALX_BUILD_CPP_DOCS`, and then create a second `MATERIALX_BUILD_DOCS` option that emits a deprecation warning to the user and sets `MATERIALX_BUILD_CPP_DOCS` to `ON`?

Reply:
yeah, `MATERIALX_BUILD_CPP_DOCS` would likely be clearer.

I've renamed it in https://github.com/AcademySoftwareFoundation/MaterialX/pull/2038/commits/ce55c88b6fb7042c363883b844929d97f084db82 but keeping the old name around as a deprecated option, similar to `MATERIALX_BUILD_IOS`.

Would we want to rename the `MaterialXDocs` target to `MaterialXDocsCpp` (or something like that) as well, to go with `MaterialXDocsPython`?

Reply:
I think that's a great idea!

Reply:
Done:

I've added commit https://github.com/AcademySoftwareFoundation/MaterialX/pull/2038/commits/c9ba91f809726e3f81f00910acf2aa14d1cec861 to rename `MaterialXDocs` to `MaterialXDocsCpp`. The commit also adds a new target named `MaterialXDocs` that depends on `MaterialXDocsCpp`. The new target uses a comment with a warning message:
```cmake
# Display a deprecation warning if the old name of the target is used, but
# build the target using its new name instead, by making MaterialXDocs depend
# on MaterialXDocsCpp
add_custom_target(MaterialXDocs
                  DEPENDS MaterialXDocsCpp
                  COMMENT "WARNING: The MaterialXDocs target name is deprecated. The MaterialXDocsCpp target was built instead.")
```
When building the C++ API docs using `cmake --build . --target MaterialXDocs`, the message for that target mentions that the docs were built using that deprecated target name:
```python
[...]
build.sh: Building target MaterialXDocs...
[  0%] Generating MainPage.md
[100%] Building MaterialX C++ API Documentation in /Users/stefan/Developer/MaterialX/build/documents/html/ ...
[100%] Built target MaterialXDocsCpp
[100%] WARNING: The MaterialXDocs target name is deprecated. The MaterialXDocsCpp target was built instead.
[100%] Built target MaterialXDocs
[...]
```
This way, everyone's custom build scripts and aliases (if any) would still work, but issue that warning message.